cip-0009.md

```diff
@@ -0,0 +1,74 @@
+        CIP: 9
+        Title: Enhanced Send
+        Authors: Joe Looney (loon3) & needmoney90
+        Status: Draft
+        Type: Standards Track
+        Created: 2017-02-25
+        Discussions-To: https://counterpartytalk.org/t/send-payment-id-field-cip-discussion/2689/29
+
+
+# Abstract
+
+Establishes a new send message type with two distinct features...  
+
+* Eliminate the receiving address dust output by encoding the receiving address in the Counterparty message data.  
+* User specified Payment ID (optional)
+
+
+# Motivation
+
+To additional functionality to an asset send while at the same time making asset sends cheaper and more efficient.
+
+Enhanced sends have a smaller transaction size, eliminate dust outputs, and add a payment ID that enables greater efficiency and lower tx fees for exchange operators.
+
+
+# Rationale
+
+Asset send messages use only 28 bytes of data.  There are 54 bytes available to increase send efficiency (by including the receiving address), add functionality (a 30-byte Payment ID) and encode the Counterparty message data within an OP_RETURN output.
+
+
+# Definitions
+
+Payment ID
+
+        A 30-byte sender specified identifier included as part of an asset send message
+        
+
+# Overview
+
+An Enhanced Asset Send adds the following two features to a traditional Asset Send transaction...
+
+### Encoding Receiving Address in Message Data
+
+An asset send transaction currently requires a P2PKH output to the receiver of an asset.  This is in addition to the message data output (encoded in OP_RETURN).  The additional output is an unnecessary burden that, on average, is *five times* more expensive to spend than it's worth (using an avg fee of 200 sat/byte and 148 bytes per tx input).  At a bitcoin price of $1200, that's a cost of $0.37 (29600 sat) to spend a $0.07 (5430 sat) dust output!
+
+### Payment ID 
+
+The Payment ID is usually used to identify transactions to merchants and exchanges: when a single public address is used for incoming transactions, the Payment ID is especially useful to tie incoming payments with user accounts. 
+
+
+# Specification
+
+The enhanced send, detailed in this proposal, utilizes the existing asset send message data structure with two additional elements: a receiving address and a Payment ID.  The total enhanced send message data size is 80 bytes to allow for storage within an OP_RETURN output.
+
+*   New Message Type (ID=2)
+*   Message Structure (CNTRPRTY + Message ID + Asset Name + Asset Qty + Receiving Address + Payment ID)
+*   Receiving Address, 21 bytes (Base58->Hex, no checksum). 
+    For converting address from Base58->Hex and vice versa, see http://lenschulwitz.com/base58
+*   Payment ID, 30 bytes (Hex) preceded by a 1 byte Payment ID length value. The Payment ID to be stored as a hex string to allow versatility for end users and developers.
```

Comment:
The 1 byte length value isn't needed here since this is the last item in the message.